driver.go

```diff
@@ -133,7 +133,10 @@ type ocConn struct {
 }
 
 func (c ocConn) Ping(ctx context.Context) (err error) {
-	defer recordCallStats(ctx, "go.sql.ping", c.options.InstanceName)(err)
+	var recordCallStatsFunc = recordCallStats(ctx, "go.sql.ping", c.options.InstanceName)
+	defer func() {
+		recordCallStatsFunc(err)
+	}()
```

Comment:
```go
    onDeferWithErr := recordCallStats(ctx, "go.sql.ping", c.options.InstanceName)
    defer func() {
          // Invoking this function in a defer so that we can capture
          // the value of err as set on function exit.
          onDeferWithErr(err)
    }()
```